### DISCUSSION TITLE
Cp and Ct calculations with Yaw in Floris/Simulation/Turbine.py

### DISCUSSION THREAD

**Discussion by RyanDunn729** [NatLabRockies/floris#618] [Q&A]
Hello, I was looking into re-creating the results generated by Floris wake models with yaw adjustments, and one important aspect was the turbine models. My question is with regards to Ct and Cp estimation of a yawed turbine.
For Cp/Power, it appears that the way that yaw is accounted for is by creating this intermediate "yaw_effective_velocity" and using that the surrogate model, see lines 141-145 and 155
```python
    yaw_effective_velocity = (
        (air_density/ref_density_cp_ct)**(1/3)
        * average_velocity(velocities)
        * cosd(yaw_angle) ** pW
    )
...
     power_interp[turb_type](yaw_effective_velocity)
```
However, a different approach for Ct to accommodate for yaw is used. Instead, the surrogate model is just directly multiplied by cos(yaw), see line 213
```python
    effective_thrust = thrust_coefficient * cosd(yaw_angle)
```

Is this a bug or is there some underlying purpose that Ct and Cp are treated differently? Intuition tells me that yaw_effective_velocity is most appropriate for both.

**Comment by paulf81** [comment]
I think you are right that in principle both could be done the way power is computed.  However, I think the reason for the difference is that we are usually interested in the power (in kW), and not Cp, but for thrust it is the other way around, with Ct in input to several wake models but thrust itself not used.  In the original FLORIS we interpolated directly on both Cp and Ct, and this led to numerical problems in above rated for power, where it was hard to keep power perfectly flat with respect to wind speed, motivating the change to interpolating on power directly.  The unflatness of the power is small, but creates problems for optimizers.  Similarly, moving to yaw_effective_velocity helps also to deal with above rated conditions.  In principle then computing Ct could be done in the same way as power, but I think in practice the ultimate change wrt final result would be small so I think that is why we've kept it simpler.  But perhaps this should be reconsidered?
